numerrin_wrapper/numerrin_code.py

```diff
@@ -0,0 +1,159 @@
+        numerrin.parsestring(self.ph, self.ch, codeString)
+
+    def execute(self, nproc):
+        """ execute Numerrin code
+
+        Parameters
+        ----------
+        nproc : int
+            number of processor cores
+
+        """
+        numerrin.execute(self.ph, self.ch, nproc)
+
+    def clear(self):
+        """ clear Numerrin code
+
+        """
+        numerrin.clearcode(self.ch)
+
+    def generate_code(self, CM, SP, BC, CMExt):
+        """ generate Numerrin code according to user settings
+
+        Parameters
+        ----------
+        CM : DataContainer
+            Computational Method
+        SP : DataContainer
+            System Parameters
+        BC : DataContainer
+            Boundary Conditions
+        CMExt : dictionary
+            extension to CM
+
+        Return
+        ------
+        code : str
+            Numerrin code as a string
+
+        """
+
+        name = CM[CUBA.NAME]
+        code = ""
+
+        funccode = ""
+        # define spaces
+        spacecode = "V=Space(omega,\"Lagrange\",1)\n"
+        spacecode += "W=Space(omega,\"Lagrange\",1)\n"
+        # define associations
+        assoccode = "u[0:2] in V\n"
+        assoccode += "p in W\n"
+        assoccode += "q[0:2]->u\n"
+        assoccode += "q[3]->p\n"
+        assoccode += "r:=q\n"
+        # define domains and boundary conditions
+        domaincode = ""
+        bccode = ""
+        pressureBCs = BC[CUBA.PRESSURE]
+        nonFixedBoundaryTypes = ["zeroGradient",
+                                 "fixedFluxPressure",
+                                 "empty"]
+        for boundary in pressureBCs:
+            if pressureBCs[boundary] not in nonFixedBoundaryTypes:
+                bi = int(boundary.replace('boundary', ''))
+                domaincode += boundary + "->" + name + "domains" +\
+                    str(bi) + "\n"
+                bccode += "Constraint(" + boundary + ",W)\n"
+                bccode += " r[3] <- p(.) - " +\
+                    str(pressureBCs[boundary]) +\
+                    "\n"
+                bccode += "EndConstraint\n"
+
+        velocityBCs = BC[CUBA.VELOCITY]
+        for boundary in velocityBCs:
+            if velocityBCs[boundary] not in nonFixedBoundaryTypes:
+                bi = int(boundary.replace('boundary', ''))
+                velo = velocityBCs[boundary]
+                domaincode += boundary + "->" + name + "domains" +\
+                    str(bi) + "\n"
+                bccode += "Constraint(" + boundary + ",V)\n"
+                bccode += " up=u(.)\n"
+                bccode += " r[0] <- up[0] - " + str(velo[0]) + "\n"
+                bccode += " r[1] <- up[1] - " + str(velo[1]) + "\n"
+                bccode += " r[2] <- up[2] - " + str(velo[2]) + "\n"
+                bccode += "EndConstraint\n"
+
+        GE = CMExt[CUBAExt.GE]
+        if CUBAExt.LAMINAR_MODEL in GE:
+            solver = "stabilizedLaminarNS3D"
+            if solver == 'stabilizedLaminarNS3D':
```

Comment:
What is the purpose of this conditional statement (variable assigned on the above line)?